### PR TITLE
beego.ParseForm() improvement

### PR DESCRIPTION
**Pull request by Zaaksam**
Support for anonymous struct

### CODE REVIEW

File: templatefunc.go [hunk 1]
```diff
@@ -280,22 +280,23 @@ func AssetsCSS(src string) template.HTML {
...
-	objT := reflect.TypeOf(obj)
-	objV := reflect.ValueOf(obj)
-	if !isStructPtr(objT) {
-		return fmt.Errorf("%v must be  a struct pointer", obj)
-	}
-	objT = objT.Elem()
-	objV = objV.Elem()
-
+// Support for anonymous struct.
+func parseFormToStruct(form url.Values, objT reflect.Type, objV reflect.Value) error {
```
Comment (astaxie) [inline templatefunc.go]: why add a new function `parseFormToStruct`, it looks still the same logic for ParseForm. you just add a Anonymous

Comment (Zaaksam) [inline templatefunc.go]: it seems inelegant, i just want to support the multi-nested anonymous struct

**Comment by astaxie** [comment]
@Zaaksam could you  add same test case for it?

**Comment by Zaaksam** [comment]
Already added

**Comment by astaxie** [comment]
awesome. thx

**Pull request merged**